docs/data/tree-view/rich-tree-view/selection/SelectionPropagation.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import {
+  TreeViewBaseItem,
+  TreeViewSelectionPropagation,
+} from '@mui/x-tree-view/models';
+
+const MUI_X_PRODUCTS: TreeViewBaseItem[] = [
+  {
+    id: 'grid',
+    label: 'Data Grid',
+    children: [
+      { id: 'grid-community', label: '@mui/x-data-grid' },
+      { id: 'grid-pro', label: '@mui/x-data-grid-pro' },
+      { id: 'grid-premium', label: '@mui/x-data-grid-premium' },
+    ],
+  },
+  {
+    id: 'pickers',
+    label: 'Date and Time Pickers',
+    children: [
+      { id: 'pickers-community', label: '@mui/x-date-pickers' },
+      { id: 'pickers-pro', label: '@mui/x-date-pickers-pro' },
+    ],
+  },
+  {
+    id: 'charts',
+    label: 'Charts',
+    children: [{ id: 'charts-community', label: '@mui/x-charts' }],
+  },
+  {
+    id: 'tree-view',
+    label: 'Tree View',
+    children: [{ id: 'tree-view-community', label: '@mui/x-tree-view' }],
+  },
+];
+
+export default function SelectionPropagation() {
+  const [selectionPropagation, setSelectionPropagation] =
+    React.useState<TreeViewSelectionPropagation>({
+      parents: true,
+      descendants: true,
+    });
+
+  return (
+    <div style={{ width: '100%' }}>
+      <Stack direction="row" spacing={2}>
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={selectionPropagation.descendants}
+              onChange={(event) =>
+                setSelectionPropagation((prev) => ({
+                  ...prev,
+                  descendants: event.target.checked,
+                }))
+              }
+            />
+          }
+          label="Auto select descendants"
+        />
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={selectionPropagation.parents}
+              onChange={(event) =>
+                setSelectionPropagation((prev) => ({
+                  ...prev,
+                  parents: event.target.checked,
```

Comment:
Clicking on an indeterminate checkbox when `selectionPropagation.parents='true'` does nothing. Is it intended?

https://github.com/user-attachments/assets/6ccfeb27-933f-4b54-9418-272ae6802057

Reply:
Probably not :laughing: 
I'll have a look

Reply:
The grid has the exact same behavior
It come from this line: https://github.com/mui/mui-x/blob/4bc5acc101590bf64eef33270ef803647e249dde/packages/x-data-grid/src/hooks/features/rowSelection/utils.ts#L94

https://github.com/user-attachments/assets/eefab1b3-073b-43a0-9caf-4e35d626be2d

Not sure if removing this condition has unplaned consequences

Reply:
Correct, the issue exists on the Data Grid side too, but there's a slight difference. It's only reproducable only when the `descendants` checkbox is not checked.

I'll try to make fix the behavior on the Data Grid side. 👍 

P.S The attached video doesn't work for me, maybe it wasn't fully uploaded?

Reply:
**Update:** I think the Data Grid's behavior is correct and intended.

It fulfills this constraint: _"When `rowSelectionPropagation.parents='true'`, a parent will (only) be selected if all it's descendants are selected."_

So, if any of the descendants of a particular parent is not selected and the parent itself is allowed to be selected it'd violate the above constraint.

But in combined with `rowSelectionPropagation.descendants='true'`, it should work fine (i.e. select the parents + all the descendants)

Does it make sense?

Reply:
Fixed
The Tree View was not handling correctly the scenario where the checkbox is checked but the item is not in the model (scenario introduced in this PR with the `indeterminate` state).

I do agree with you, it's not a great UX but I suppose it highlights that most people should enable both parents and descendants selection.